unified_planning/model/problem.py

```diff
@@ -558,20 +569,67 @@ def add_goal(
         if goal_exp != self._env.expression_manager.TRUE():
             self._goals.append(goal_exp)
 
+    def add_trajectory_constraint(self, constraint: "up.model.fnode.FNode"):
+        """
+        Adds the given `trajectory_constraint` to the `Problem`;
+        a trajectory_constraint is an expression defined as:
+            Always, Sometime, At-Most-Once, Sometime-Before, Sometime-After or
+            defined with quantifiers.
+
+        :param trajectory_constraint: The expression added to the `Problem`.
+        """
+        new_traj_list = []
+        simplifier = up.model.walkers.simplifier.Simplifier(self._env, self)
+        (constraint_exp,) = self._env.expression_manager.auto_promote(constraint)
+        new_traj_list.append(
+            simplifier.simplify(
+                self._env.expression_manager.And(
+                    self._trajectory_constraints,
+                    constraint_exp,
+                )
+            )
+        )
+        self._trajectory_constraints = new_traj_list
```

Comment:
Here it is a bit ugly to create a `Simplifier` and use it for just 1 simplification. Also, this may lead to unwanted behaviors, mainly because the `Simplifier` with a problem at the constructor also simplifies static fluents. But, an Action that modifies said fluent might be added later, thus making this simplification actually wrong.

Is the simplification here really needed?
We don't simplify `goals` etc. Also, here we are missing the check that it is a `trajectory_constraint`.

If it is, I would do it this way:
(Note that you lose simplification over static fluents)
```suggestion
        new_traj_list = []
        (constraint_exp,) = self._env.expression_manager.auto_promote(constraint)
        all_constraints_simplified = self._env.expression_manager.And(
            constraint_exp, 
            *self._trajectory_constraints
        ).simplify()
        self._trajectory_constraints = all_constraints_simplified.args.copy() if all_constraints_simplified.is_and() else [all_constraints_simplified]
```